Userland/Services/ChessEngine/ChessEngine.cpp

```diff
@@ -68,3 +68,9 @@ void ChessEngine::handle_quit()
     if (on_quit)
         on_quit(ESUCCESS);
 }
+
+void ChessEngine::handle_unexpected_eof()
+{
+    if (on_quit)
+        on_quit(EPIPE);
```

Comment:
I'm not sure I understand. This code doesn't change after it is introduced.

Reply:
That's partially due to me not paying attention to the fact that it was not in `LibChess`, my bad.
It's however better to have put it in its own commit.

Reply:
> It's however better to have put it in its own commit.

I'm assuming this is resolved now? I think I might have messed up the commit history at some point and squashed two commits together.

Otherwise I'm not certain how I could further separate any commits.

Reply:
It's fine as is, you can also merge [ChessEngine: Gracefully handle GUI disconnections](https://github.com/SerenityOS/serenity/pull/18140/commits/89e494110fa6281b622d55bfb1a6699ef55ecb53) in [LibChess: Allow UCIEndpoint to handle unexpected disconnections](https://github.com/SerenityOS/serenity/pull/18140/commits/04066a83689bdd4b79036d979e8dccfa2d9288af)

Reply:
I've kept changes to `LibChess`, `ChessEngine` and `Chess` separate. Should I not do that? I could merge everything into 2 commits like this:

`LibChess+ChessEngine+Chess: Implement the UCI quit command`
`LibChess+ChessEngine+Chess: Gracefully handle disconnections`

Would that be better?

Reply:
Nah forget it, it's fine as is.